connector_magento/components/importer.py

```diff
@@ -0,0 +1,324 @@
+        record_ids = self.backend_adapter.search(filters)
+        for record_id in record_ids:
+            self._import_record(record_id)
+
+    def _import_record(self, external_id):
+        """ Import a record directly or delay the import of the record.
+
+        Method to implement in sub-classes.
+        """
+        raise NotImplementedError
+
+
+class DirectBatchImporter(AbstractComponent):
+    """ Import the records directly, without delaying the jobs. """
+
+    _name = 'magento.direct.batch.importer'
+    _inherit = 'magento.batch.importer'
+
+    def _import_record(self, external_id):
+        """ Import the record directly """
+        self.model.import_record(self.backend_record, external_id)
+
+
+class DelayedBatchImporter(AbstractComponent):
+    """ Delay import of the records """
+
+    _name = 'magento.delayed.batch.importer'
+    _inherit = 'magento.batch.importer'
+
+    def _import_record(self, external_id, job_options=None, **kwargs):
+        """ Delay the import of the records"""
+        delayable = self.model.with_delay(**job_options or {})
+        delayable.import_record(self.backend_record, external_id, **kwargs)
+
+
+class SimpleRecordImporter(Component):
+    """ Import one Magento Website """
+
+    _name = 'magento.simple.record.importer'
+    _inherit = 'magento.importer'
+    _apply_on = [
+        'magento.res.partner.category',
+    ]
+
+
+class TranslationImporter(Component):
+    """ Import translations for a record.
+
+    Usually called from importers, in ``_after_import``.
+    For instance from the products and products' categories importers.
+    """
+
+    _name = 'magento.translation.importer'
+    _inherit = 'magento.importer'
+    _usage = 'translation.importer'
+
+    def _get_magento_data(self, storeview_id=None):
+        """ Return the raw Magento data for ``self.external_id`` """
+        return self.backend_adapter.read(self.external_id, storeview_id)
+
+    def run(self, external_id, binding, mapper=None):
+        self.external_id = external_id
+        storeviews = self.env['magento.storeview'].search(
+            [('backend_id', '=', self.backend_record.id)]
+        )
+        default_lang = self.backend_record.default_lang_id
+        lang_storeviews = [sv for sv in storeviews
+                           if sv.lang_id and sv.lang_id != default_lang]
+        if not lang_storeviews:
+            return
+
+        # find the translatable fields of the model
+        fields = self.model.fields_get()
+        translatable_fields = [field for field, attrs in list(fields.items())
+                               if attrs.get('translate')]
+
+        if mapper is None:
+            mapper = self.mapper
+        else:
+            mapper = self.component_by_name(mapper)
+
+        for storeview in lang_storeviews:
+            lang_record = self._get_magento_data(storeview.external_id)
+            map_record = mapper.map_record(lang_record)
+            record = list(map_record.values())
```

Comment:
Don't convert this dictionary of values to `list`. You can tell from line 320 that `record` needs to stay in dictionary format so that you can fetch its `items()`